model/trace/instrumentation/gcp-client.yml

```diff
@@ -0,0 +1,43 @@
+groups:
+  - id: gcp.client.scope
+    type: scope
+    brief: >
+      The `gcp.client` conventions are intended to be used by Google Cloud
+      client libraries (e.g. https://github.com/googleapis/* and other
+      Google-owned/Google-provided client libraries for Google Cloud) to
+      identify such clients and to track their operations and performance.
+    prefix: gcp.client
+    stability: experimental
+    attributes:
+      - id: gcp.client.service
+        type: string
+        stability: experimental
+        requirement_level: required
+        brief: >
+          Short name of the GCP service for which the library is a client.
+          All variants of the client across programming languages should use an
+          identical value for this property. The short name should be intuitive
+          and aligned with either the name of the client (e.g. X for client
+          library packages named `google-cloud-X`) or the name of the API
+          subdomain (e.g. X for APIs hosted at `X.googleapis.com`). 
+        examples: ['cloudsql', 'datastore', 'spanner', 'firestore']
+      - id: gcp.client.repo
+        type: string
+        stability: experimental
+        brief: >
+          Identifies the repository where the client library source comes from.
+          The prefix `github.com/` should be not be included in the name. If 
+          (`gcp.client.service`, `telemetry.sdk.language`) do not uniquely
+          identify the client such as when multiple client variants exist
+          for the same service in a given programming language, this field
+          should be supplied in order to disambiguate between clients.
+      - id: gcp.client.artifact
```

Comment:
what's the relationship between instrumentation scope name and the artifact name?

We used to think about instrumentation scope name as an artifact name, but it's not a requirement  (and not a right level of granularity), I believe we miss a general-purpose attribute to identify the artifact name and the repository. there is nothing specific to GCP here.

I'd suggest adding attributes like `instrumentation.artifact.name` and `instrumentation.artifact.repo` (in case of instrumentation library, they should match the instrumentation library name, in case of native instrumentation, they should match the name of the original library).

Reply:
Thanks! I like the suggestion of making these into general scope properties.

As for "name", there seemed to be some variability into how it was being interpreted/set.

Reply:
> As for "name", there seemed to be some variability into how it was being interpreted/set.

agreed, we can consider `full_name` or something similar, suggest to use whatever is idiomatic in specific language/ecosystem and add a few examples (e.g. `com.google.cloud:google-cloud-storage`, `@azure/storage-blob`, `opentelemetry-instrumentation-boto`)

Reply:
I've updated this code based on your suggestions to generalize. I ended up at `instrumentation.source.` instead of `instrumentation.artifact.`, because the intention was to describe the original source code more so than where it is being packaged up and published. However, I think this falls in line with where you were suggesting/heading.

To summarize, the new attributes are:

  - `gcp.client.service`
  -  `instrumentation.source.repo`
  -  `instrumentation.source.target`

I made one slight semantic change with `repo`; previously, I had it omit the domain (because GCP client libraries are always published on GitHub). With this generalization, I felt that it was appropriate to include the `github.com/` prefix so as to make this more generalizable to code which may be hosted elsewhere.

Reply:
I've moved the general, vendor-agnostic conventions to a separate pull request:

  - https://github.com/open-telemetry/semantic-conventions/pull/1094

Let's continue the conversation there.

In the meantime, I hope we can make progress on and get in `gcp.client.service` (now the sole focus of this PR).

Thanks!